### PR TITLE
Redmine#60 Add wrapper functions for linux user and group management

### PR DESCRIPTION
**Pull request by earl-jann**
Redmine#60 Add wrapper functions for linux user and group management

  * initial commit for the modules below
  * Added user management modules (add,delete,modify,get)
  * Added group management modules (add,delete,modify,get)
  * TODO: code cleanup, promises, directory structure

### CODE REVIEW

File: node_modules/linux/package.json [hunk 1]
```diff
@@ -0,0 +1,25 @@
+{
+  "name": "lohika-linux",
+  "version": "0.1.0",
+  "description": "linux nodejs modules for lohika",
+  "main": "index.js",
```
Comment (joegen) [inline node_modules/linux/package.json]: Is index.js the convention used generally for module entry points?  This feels too HTML'ish for me

File: node_modules/linux/src/getgroups.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+module.exports = function getgroups(options, callback) {
+    // if it's a pure string, use as is
+  var args = ['-c', 'getent group | awk -F\':\' \'{ print $1}\''];
+
+  // console.log('spawning childProcess with args: [' + args + ']');
+
+  var childProcess = child_process.spawn('sh', args);
+
+  // console.log('childProcess spawned: ' + childProcess);
```
Comment (joegen) [inline node_modules/linux/src/getgroups.js]: Please remove entirely if not needed

**Comment by earl-jann** [comment]
I was thinking of replacing the "console.log" with a debug logger module..
will remove them instead

**Comment by joegen** [comment]
Closing this pull request.  Please submit the new one with all the changes required

File: node_modules/linux/src/getgroups.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    groups = groups.concat(data.split('\n'));
+  });
+
+  childProcess.on('exit', function(code, signal) {
+    // console.log('error: ' + error);
+    // console.log('exit code: ' + code);
+    var error; 
+
+    if (code) {
+      error = new Error('Invalid exit code: ' + code);
```
Comment (earl-jann) [inline node_modules/linux/src/getgroups.js]: done

**Pull request closed without merging**